lib/build_manager.py

```diff
@@ -33,20 +33,17 @@ def __init__(self, packages_names, distro):
 
     def __call__(self):
         try:
-            self.prepare_packages()
-
+            error_number = super(BuildManager, self).__call__()
         # distro related issues
         except (exception.DistributionNotSupportedError,
                 exception.DistributionVersionNotSupportedError,
                 exception.DistributionDetectionError) as exc:
             LOG.exception("Error during distribution detection. "
                           "See the logs for more information")
             return exc.errno
-        # package issues
-        except exception.PackageError as exc:
-            LOG.exception("Failed to load the package in components. "
-                          "See the logs for more information")
-            return exc.errno
+        else:
+            if error_number:
+                return error_number
 
         return self.build()
 
```

Comment:
I'm not sure which two methods you're refering to (__call__ and prepare_packages?).
I've removed the method prepare_packages from BuildManager and had to add an argument "package_class" in PackagesManager.prepare_packages, to allow creating RPM packages.

One more think that I think could be changed is the __call__ method. It's kind of confusing to call BuildManager()() and calling a named method like "build" or "prepare_packages" is more straightfoward. That can be done later, if you agree.